Sources/PackageGraph/ModuleAliasTracker.swift

```diff
@@ -221,20 +254,38 @@ class ModuleAliasTracker {
                             // If there are no aliases or conflicting aliases
                             // for the same key defined in other child products,
                             // those aliases should be removed from this target.
-                            let hasConflict = allTargetsInOtherChildProducts.contains{ otherTarget in
+                            let targetsWithConflictingAliases = allTargetsInOtherChildProducts.filter { otherTarget in
                                 if let otherAlias = otherTarget.moduleAliases?[nameToBeAliased] {
-                                    return otherAlias != aliasInChild
+                                    return otherAlias != aliasInCurChild
                                 } else {
                                     return otherTarget.name == nameToBeAliased
                                 }
                             }
-                            if hasConflict {
-                                // If there are aliases, remove as aliasing should
-                                // not be applied
+
+                            if !targetsWithConflictingAliases.isEmpty {
+                                // If there are conflicting aliases, remove them
+                                // and show a diag message
+                                let conflictingAliases = targetsWithConflictingAliases.compactMap { $0.moduleAliases?[nameToBeAliased] }
+                                var message = conflictingAliases.isEmpty ? "Targets named '\(nameToBeAliased)' found in other dependency products;" : "Conflicting module aliases found \(conflictingAliases + [aliasInCurChild]) for '\(nameToBeAliased)'; "
+                                message += "these will be removed from target '\(curTarget.name)'"
+                                observabilityScope.emit(info: message)
```

Comment:
is this informational, or should be a warning?

Reply:
this is info; changed wording